media-libs/libsdl2/patches/libsdl2-2.0.20.patchset

```diff
@@ -1267,143 +1267,532 @@ index ca9dcad..1ac3201 100644
+     SDL_free(device);
+ }
+ 
+@@ -273,7 +282,7 @@ static void HAIKU_MouseInit(_THIS)
+ int HAIKU_VideoInit(_THIS)
+ {
+     /* Initialize the Be Application for appserver interaction */
+-    if (SDL_InitBeApp() < 0) {
++    if (SDL_InitBeApp(_this) < 0) {
+         return -1;
+     }
+     
+diff --git a/src/video/haiku/SDL_bvideo.h b/src/video/haiku/SDL_bvideo.h
+index d0d7bfe..879a886 100644
+--- a/src/video/haiku/SDL_bvideo.h
++++ b/src/video/haiku/SDL_bvideo.h
+@@ -22,6 +22,66 @@
+ #ifndef BVIDEO_H
+ #define BVIDEO_H
+ 
++/* Forward declaration */
++class SDL_BWin;
++
++#include <vector>
++
++typedef struct SDL_VideoData {
++	std::vector<SDL_Window*> _window_map;
++	    /* Window creation/destruction methods */
++    int32 GetID(SDL_Window *win) {
++        int32 i;
++        for(i = 0; i < _GetNumWindowSlots(); ++i) {
++            if( GetSDLWindow(i) == NULL ) {
++                _SetSDLWindow(win, i);
++                return i;
++            }
++        }
++        /* Expand the vector if all slots are full */
++        if( i == _GetNumWindowSlots() ) {
++            _PushBackWindow(win);
++            return i;
++        }
++
++        /* TODO: error handling */
++        return 0;
++    }
++	void ClearID(SDL_BWin *bwin);
++    SDL_Window *GetSDLWindow(int32 winID) {
++        return _window_map[winID];
++    }
++	bool _GetWinID(BMessage *msg, int32 *winID) {
++        return msg->FindInt32("window-id", winID) == B_OK;
++    }
++	void _SetSDLWindow(SDL_Window *win, int32 winID) {
++        _window_map[winID] = win;
++    }
++	int32 _GetNumWindowSlots() {
++        return _window_map.size();
++    }
++	void _PopBackWindow() {
++        _window_map.pop_back();
++    }
++	void _PushBackWindow(SDL_Window *win) {
++        _window_map.push_back(win);
++    }
 +#if SDL_VIDEO_OPENGL
-+BGLView* SDL_BApp::_current_context = NULL;
++    BGLView *GetCurrentContext() {
++        return _current_context;
++    }
++    
++    void SetCurrentContext(BGLView *newContext) {
++        if(_current_context)
++            _current_context->UnlockGL();
++        _current_context = newContext;
++        if (_current_context)
++            _current_context->LockGL();
++    }
++	BGLView *_current_context;
 +#endif
++} SDL_VideoData;
 +
  #ifdef __cplusplus
  extern "C" {
  #endif
+@@ -29,7 +89,6 @@ extern "C" {
+ #include "../../main/haiku/SDL_BeApp.h"
+ #include "../SDL_sysvideo.h"
+ 
+-
+ extern void HAIKU_VideoQuit(_THIS);
+ extern int HAIKU_VideoInit(_THIS);
+ extern void HAIKU_DeleteDevice(_THIS);
+diff --git a/src/video/haiku/SDL_bwindow.cc b/src/video/haiku/SDL_bwindow.cc
+index 587affa..ca18c62 100644
+--- a/src/video/haiku/SDL_bwindow.cc
++++ b/src/video/haiku/SDL_bwindow.cc
+@@ -71,7 +71,7 @@ static int _InitWindow(_THIS, SDL_Window *window) {
+         return -1;
+ 
+     window->driverdata = bwin;
+-    int32 winID = _GetBeApp()->GetID(window);
++    int32 winID = _GetBeApp()->driverdata->GetID(window);
```

Comment:
This ought to be _this->driverdata... but I had to cast driverdata to an SDL_VideoData* to get it to compile. Looking in SDL_sysvideo.h driverdata is defined as void* so that is no great surprise. Many backends call the custom struct type they use for this SDL_VideoData, so that part is ok at least.